### ISSUE TITLE
Emacs shell incorrectly doesn't get the fancy output

### ISSUE THREAD

**Issue by fitzgen** [zkat/miette#93]
Thanks for creating `miette`! :)

### Steps to Reproduce

Add the example program from the `README` to `src/main.rs`:

<details><summary><code>src/main.rs</code></summary>

```rust
/*
You can derive a Diagnostic from any `std::error::Error` type.

`thiserror` is a great way to define them, and plays nicely with `miette`!
*/
use miette::{Diagnostic, SourceSpan};
use thiserror::Error;

#[derive(Error, Debug, Diagnostic)]
#[error("oops!")]
#[diagnostic(
    code(oops::my::bad),
    url(docsrs),
    help("try doing it better next time?"),
)]
struct MyBad {
    // The Source that we're gonna be printing snippets out of.
    // This can be a String if you don't have or care about file names.
    #[source_code]
    src: NamedSource,
    // Snippets and highlights can be included in the diagnostic!
    #[label("This bit here")]
    bad_bit: SourceSpan,
}

/*
Now let's define a function!

Use this Result type (or its expanded version) as the return type
throughout your app (but NOT your libraries! Those should always return concrete
types!).
*/
use miette::{Result, NamedSource};
fn this_fails() -> Result<()> {
    // You can use plain strings as a `Source`, or anything that implements
    // the one-method `Source` trait.
    let src = "source\n  text\n    here".to_string();

    Err(MyBad {
        src: NamedSource::new("bad_file.rs", src),
        bad_bit: (9, 4).into(),
    })?;

    Ok(())
}

/*
Now to get everything printed nicely, just return a Result<()>
and you're all set!

Note: You can swap out the default reporter for a custom one using `miette::set_hook()`
*/
fn main() -> Result<()> {
    // kaboom~
    this_fails()?;

    Ok(())
}
```

</details>

And this `Cargo.toml`:

<details><summary><code>Cargo.toml</code></summary>

```toml
[package]
name = "miette-test-case"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
miette = { version = "3.2.0", features = ["fancy"] }
thiserror = "1.0.30"
```

</details>

Now open Emacs, and then open the Emacs shell via `M-x shell`.

Finally, run the command `cargo run` within the Emacs shell.

### Expected Results

The fancy output is enabled, and we see the expected source snippets and unicode box drawing characters underlining the source spans and all that.

Basically the same output that you get running `cargo run` in a regular terminal:

![Screenshot from 2021-11-15 14-39-37](https://user-images.githubusercontent.com/74571/141864573-32066cdf-755a-4501-b1ca-1ab59d3c05b4.png)

### Actual Results

We do not get the fancy output, and seem to get a screen reader compatible output that is hard to read instead:

![Screenshot from 2021-11-15 14-40-01](https://user-images.githubusercontent.com/74571/141864633-9bb59c70-c0e0-4267-a63b-0b13893b5030.png)

**Comment by fitzgen** [comment]
For what it's worth, this can be added to the top of `main` as a temporary work around:

```rust
        let _ = miette::set_hook(Box::new(|_| {
            Box::new(
                miette::MietteHandlerOpts::new()
                    .force_graphical(true)
                    .build(),
            )
        }));
```

**Comment by zkat** [comment]
The code controlling this is over here:

https://github.com/zkat/miette/blob/main/src/handler.rs#L186-L196

and by extension,

https://github.com/zkat/supports-color/blob/main/src/lib.rs

There could also probably be some logic around "well, it doesn't support color, but it does support some level of graphical output", but I made a design decision at some point that "no color" in miette means "screen reader friendly" so you could just use NO_COLOR to make it narratable. That may have been a mistake? I'm happy to change that as long as there's an easy out for screen reader users.

**Comment by fitzgen** [comment]
For what its worth, the Emacs shell also supports (at least most?) ansi colors. But it sets `TERM=dumb` by default, because it doesn't support many of the more advanced control codes for moving the cursor around and clearing output and things like that.

**Comment by zkat** [comment]
Should I stop punning off NO_COLOR for accessible output?

**Comment by fitzgen** [comment]
I'm not sure, unfortunately. I don't have a lot of experience with terminal feature detection.

**Comment by fitzgen** [comment]
Emacs will also automatically set the `INSIDE_EMACS` env var. Perhaps worth special casing that in the handler or in `supports-color`?
